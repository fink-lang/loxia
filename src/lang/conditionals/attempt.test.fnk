{fink2js} = import '../../testing'
{describe, it, expect, to_match_snapshot} = import '../../testing/jest'


describe:: 'attempt', fn:
  it:: 'compiles', fn:

    expect::
      fink2js`
        foo1 = fn:
          [err, item] = attempt:
            bar = shrub()
            bar + ni()
          [item, err]

        foo2 = fn: attempt: throw new Error:: 'foobar'
      `
      to_match_snapshot



