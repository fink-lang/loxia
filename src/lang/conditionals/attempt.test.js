import {fink2js} from '../../testing';


test('compiles attempt', ()=> {
  expect(
    fink2js(`
      foo1 = fn:
        [item, err] = attempt:
          bar = shrub()
          bar + ni()
        [item, err]

      foo2 = fn: attempt: 1/0
    `)
  ).toMatchSnapshot();
});


