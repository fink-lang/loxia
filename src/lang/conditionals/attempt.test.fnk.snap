// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attempt compiles 1`] = `
"const foo1 = () => {
  let _do_result;

  try {
    const bar = shrub();
    _do_result = [null, bar + ni()];
  } catch (ˆerror_1) {
    _do_result = [ˆerror_1, null];
  }

  const [err, item] = _do_result;
  _do_result = undefined;
  return [item, err];
};

const foo2 = () => try {
  return [null, throw new Error(\`foobar\`)];
} catch (ˆerror_2) {
  return [ˆerror_2, null];
};

Object.assign(module.exports, {
  foo1,
  foo2
});"
`;
